### ISSUE TITLE
Should 'genome sys gateway detach' allow unmounting without deleting local files? 

### ISSUE THREAD

**Issue by malachig** [genome/gms#27]
This behavior was slightly non intuitive to me.

I am inside a VM and noticed that while everything is working smoothly, any attempt to list hidden files in /opt/gms hangs indefinitely.

ls works fine, but 'ls -a' never returns a result.  I suspected that this might be due to this mount:

curlftpfs#ftp://genome.wustl.edu/pub/software/gms/testdata/GMS1/  7.5T     0  7.5T   0% /opt/gms/.GMS1.ftp

Perhaps any 'ls' that attempts to query '/opt/gms/.GMS1.ftp' suffers from the flakiness of curlftpfs.  Or our FTP server is down (regularly true).  To test this experiment I thought I would just detach GMS1.  Unfortunately, this seems to have deleted /opt/gms/GMS1.  Or at least removed the symlink '/opt/gms/GMS1' to '/opt/gms/.GMS1.rsync/'.

What if I want to leave that data there but unmount the curlftpfs mount?

Even after detaching GMS1, the mount remains.  

With the unreliability of our FTP server and clunky filesystems like curlftpfs I would like to attach, sync the data over, and once I have a local copy, remove the mount.

Basically I don't entirely follow what is going on with these things:
/opt/gms/.GMS1.ftp/
/opt/gms/.GMS1.rsync/

Is it expected that I would have both?

**Comment by malachig** [comment]
It looks like when the user runs

```
genome sys gateway attach GMS1 --protocol ftp
```

The GMS1 data on our FTP server is mounted at '/opt/gms/.GMS1.ftp/' and a symlink '/opt/gms/GMS1/' is created to point to this.  In this setup, attempts to use the GMS1 data will happen over the FTP (via a curlftpfs mount).  We generally find that the performance of this setup is very unreliable.  We therefore recommend that instead of the above command, the user does this instead:

```
genome sys gateway attach GMS1 --protocol ftp --rsync
```

When this command is run, the same mount and symlink described above is created and an rysnc job to copy all data from the FTP to local disk begins.  This data is stored in '/opt/gms/.GMS1.rsync/'.  Once this sync is complete the '/opt/gms/GMS1/' symlink is moved from '/opt/gms/.GMS1.ftp/' to '/opt/gms/.GMS1.rsync/'.  From that point on, attempts to use or list the GMS1 data will hit your local copy and performance will be much better.

If you kill the 'genome sys gateway attach' command while syncing is ongoing, the symlink will remain pointing at the FTP mount instead of your local copy.  If you restart the command and let it finish this will be fixed.  Or you can move the symlink yourself.  Either way the curlftpfs mount will remain.  I'm still not entirely sure what the use case for a 'genome sys gateway detach' would be.   However, if you simply want to get rid of the ftp mount you can do this:

``` bash
cd /opt/gms
sudo umount /opt/gms/.GMS1.ftp
sudo rm -fr .GMS1.ftp/
```

There is no real need to do this though.  Since the curlftpfs is not placed in /etc/fstab, it will also disappear on a reboot of the host.

**Comment by malachig** [comment]
I think above description adequately explains what it going on here now.  Closing for now.
